src/components/Tooltip/Tooltip.tsx

```diff
@@ -28,45 +29,65 @@ export const Tooltip = (props: TTooltipProps): JSX.Element | null => {
     }
   }, [id])
 
-  const onMouseEnter = () => setIsHovered(true)
+  const onMouseEnter = () => {
+    setIsHovered(true)
+  }
   const onMouseLeave = () => setIsHovered(false)
 
   useHideOnScroll(onMouseLeave)
 
   const { tooltipStyles, tooltipPosition } = useGetTooltipStyles({
     elemRef: parent,
     tooltipRef: tooltipRef.current,
-    initialPosition: position
+    initialPosition: position,
+    toolTipCurrentWidth: toolTipCurrentWidth
   })
 
   useEffect(() => {
     if (parent) {
       parent.addEventListener('mouseenter', onMouseEnter)
       parent.addEventListener('mouseleave', onMouseLeave)
     }
+
+    return () => {
+      parent?.removeEventListener('mouseenter', onMouseEnter)
+      parent?.removeEventListener('mouseleave', onMouseLeave)
+    }
   }, [parent])
 
+  useEffect(() => {
+    if (isHovered) {
+      if (typeof tooltipRef?.current?.clientWidth === 'number') {
+        setToolTipCurrentWidth(tooltipRef?.current?.clientWidth)
+      }
+    } else {
+      setToolTipCurrentWidth(0)
+    }
+  }, [isHovered])
+
+  if (!isHovered) {
+    return null
+  }
+
   return (
     <>
-      {isHovered && (
-        <div
-          style={tooltipStyles}
-          data-id={dataId}
-          className={classNames(`tooltip tooltip--${size} tooltip--${tooltipPosition}`, className)}
-          ref={tooltipRef}
+      <div
+        style={tooltipStyles}
+        data-id={dataId}
+        className={classNames(`tooltip tooltip--${size} tooltip--${tooltipPosition}`, className)}
+        ref={tooltipRef}
+      >
+        <Text
+          dataId={`${dataId}-text`}
+          className="tooltip__inner"
+          type="primary"
+          weight="regular"
+          lineHeight="small"
+          size={`${size == 'small' ? 'xsmall' : 'small'}`}
         >
-          <Text
-            dataId={`${dataId}-text`}
-            className="tooltip__inner"
-            type="primary"
-            weight="regular"
-            lineHeight="small"
-            size={`${size == 'small' ? 'xsmall' : 'small'}`}
-          >
-            {text}
-          </Text>
-        </div>
-      )}
+          {text}
+        </Text>
+      </div>
       {children}
```

Comment:
I think children prop is redundant. 
If in some cases we pass children it will not be rendered because we have check above (return null  when isHovered is false).